docs/configuration-options/index.md

```diff
@@ -2851,6 +2851,103 @@ Whether to collect performance timings. When used from the command line or a con
 
 For each key, the first number represents the elapsed time while the second represents the change in memory consumption, and the third represents the total memory consumption after this step. The order of these steps is the order used by `Object.keys`. Top level keys start with `#` and contain the timings of nested steps, i.e. in the example above, the 698ms of the `# BUILD` step include the 538ms of the `## parse modules` step.
 
+### fs
+
+|          |                    |
+| -------: | :----------------- |
+|    Type: | `RollupFsModule`   |
+| Default: | `node:fs.promises` |
+
+If you want to use a custom file system module, you can set this option to an object that implements the same API as the `RollupFsModule` interface. This is useful if you want to use a different file system implementation, such as `memfs` or `browserfs`, or if you want to mock the file system for testing purposes.
+
+```typescript
+interface RollupFsModule {
+	appendFile(
+		path: string | ArrayBuffer | ArrayBufferView,
+		data: string | ArrayBuffer | ArrayBufferView,
+		options?: {
+			encoding?: string | null;
+			mode?: number | string;
+			flag?: string;
+		}
+	): Promise<void>;
+
+	copyFile(
+		source: string | ArrayBuffer | ArrayBufferView,
+		destination: string | ArrayBuffer | ArrayBufferView,
+		mode?: number
+	): Promise<void>;
+
+	mkdir(
+		path: string,
+		options?: { recursive?: boolean; mode?: number | string }
+	): Promise<void>;
+
+	mkdtemp(
+		prefix: string,
+		options?: { encoding?: string | null }
+	): Promise<string>;
+
+	readdir(
+		path: string,
+		options?: { encoding?: string | null; withFileTypes?: false }
+	): Promise<string[]>;
+
+	readFile(
+		path: string | ArrayBuffer | ArrayBufferView,
+		options?: { encoding?: string | null; flag?: string } | string
+	): Promise<string | ArrayBuffer>;
+
+	realpath(
+		path: string | ArrayBuffer | ArrayBufferView,
+		options?: { encoding?: string | null }
```

Comment:
As in `readFile`, `options` should probably be allowed to be directly an `encoding`.